### PR TITLE
add extendPool

### PR DESCRIPTION
**Pull request by sujine2**
Admin can change rewardPerSecond manually - add extendPool function

- Admin can set rewardPerSecond
- Admin can set duration
- As soon as the admin calls the extendPool function, the information related reward is changed
- Rewards should be sent in the contract in advance

### CODE REVIEW

File: test/claim.test.ts [hunk 1]
```diff
@@ -207,4 +183,81 @@ describe('StakingPool.claim', () => {
...
+      );
+
+      const poolDataAfter = await getPoolData(testEnv);
+      const userDataAfter = await getUserData(testEnv, alice);
+
+      expect(poolDataAfter).to.be.equalPoolData(expectedPoolData_2);
+      expect(userDataAfter).to.be.equalUserData(expectedUserData_2);
+    });
+
+    it('rewardPerSecond is changed and stake and claim', async () => {
```
Comment (jaeboklee-modo) [inline test/claim.test.ts:222]: I'm not quite clear what the purpose of this test case. I think it's ok to be separated.

File: test/stake.test.ts [hunk 2]
```diff
@@ -51,8 +53,6 @@ describe('StakingPool.stake', () => {
   });
 
   context('when the pool initiated', async () => {
-    const stakeAmount = utils.parseEther('100');
-
     beforeEach('init the pool', async () => {
       await testEnv.rewardAsset.connect(deployer).faucet();
```
Comment (jaeboklee-modo) [inline test/stake.test.ts]: The `faucet` and `approve` processes appear in every context. How about putting these snippets in the top hooks of its file to reduce duplicates?

File: test/withdraw.test.ts [hunk 3]
```diff
@@ -265,5 +246,118 @@ describe('StakingPool.withdraw', () => {
...
+          amount
+        );
+        const poolDataAfter = await getPoolData(testEnv);
+        const userDataAfter = await getUserData(testEnv, alice);
+  
+        expect(poolDataAfter).eql(expectedPoolData_2);
+        expect(userDataAfter).eql(expectedUserData_2);
+      });
+
+      it('rewardPerSecond is changed and stake and withdraw', async () => {
```
Comment (jaeboklee-modo) [inline test/withdraw.test.ts:322]: I understand why you take expectedPoolData_1 and expectedUserData_1 as a parameter in expectedDataAfterStake, but I don't think that expectedDataAfterWithdraw takes the `expected User, Pool Data` former function does. 
I think If the intent was to check the scenario in which the user stakes, withdraws, and then claims, the `expectedAfter` function's parameter should be actual contract data.

**Review by jaeboklee-modo — CHANGES_REQUESTED** [review]

File: test/claim.test.ts [hunk 4]
```diff
@@ -207,4 +183,81 @@ describe('StakingPool.claim', () => {
...
+      await advanceTimeTo(await getTimestamp(tx), firstRoundStartTimestamp);
+      await testEnv.stakingPool.connect(alice).stake(amount);
+      await testEnv.rewardAsset.connect(deployer).faucet();
+      await testEnv.rewardAsset.connect(deployer).transfer(testEnv.stakingPool.address, ethers.utils.parseEther('100'));
+    });
+
+    it('rewardPerSecond is changed and claim', async () => {
+      const poolDataBefore = await getPoolData(testEnv);
+      const userDataBefore = await getUserData(testEnv, alice);
+      const tx = await testEnv.stakingPool.connect(deployer).extendPool(rewardPersecond, duration);
```
Comment (qpzm) [inline test/claim.test.ts]: Actually rewardPerSecond is not changed.

File: test/utils/expect.ts [hunk 5]
```diff
@@ -157,3 +157,23 @@ export function expectDataAfterMigrate(
...
+  userData: UserData,
+  txTimeStamp: BigNumber,
+  duration : BigNumber,
+  rewardPerSecond: BigNumber
+): [PoolData, UserData] {
+  const [newPoolData, newUserData]: [PoolData, UserData] = calculateDataAfterUpdate(
+    poolData,
+    userData,
+    txTimeStamp
+  );
```
Comment (qpzm) [inline test/utils/expect.ts:173]: `calculateDataAfterUpdate` changes the `userIndex`, but `extendPool` does not change the `userIndex`.

File: test/settings.test.ts [hunk 6]
```diff
@@ -92,4 +92,18 @@ describe('StakingPool.settings', () => {
         .withArgs(testEnv.stakingPool.address, deployer.address, RAY);
     });
   });
+
+  context('reset the pool', async () => {
```
Comment (qpzm) [inline test/settings.test.ts:96]: We also need to check the rewardPerSecond, startTimestamp, endTimestamp are changed as we want.

**Review by qpzm — APPROVED** [review]

**Pull request merged**